test/spec/mfa-challenge.js

```diff
@@ -65,6 +65,201 @@ define(function(require) {
           });
         }
       });
+
+      util.itMakesCorrectRequestResponse({
+        title: 'allows verification with autoPush true',
+        setup: {
+          status: 'mfa-challenge-sms',
+          request: {
+            uri: '/api/v1/authn/factors/smsigwDlH85L9FyQK0g3/verify?autoPush=true',
+            data: {
+              stateToken: '00rt1IY9c6Q3RVc4a2jJPbS2uAtFNWJz_d8A26KTdF',
+              passCode: '123456'
+            }
+          },
+          response: 'success'
+        },
+        execute: function (test) {
+          return test.trans.verify({
+            passCode: '123456',
+            autoPush: true
+          });
+        }
+      });
+
+      util.itMakesCorrectRequestResponse({
+        title: 'allows verification with autoPush as false',
+        setup: {
+          status: 'mfa-challenge-sms',
+          request: {
+            uri: '/api/v1/authn/factors/smsigwDlH85L9FyQK0g3/verify?autoPush=false',
+            data: {
+              stateToken: '00rt1IY9c6Q3RVc4a2jJPbS2uAtFNWJz_d8A26KTdF',
+              passCode: '123456'
+            }
+          },
+          response: 'success'
+        },
+        execute: function (test) {
+          return test.trans.verify({
+            passCode: '123456',
+            autoPush: false
+          });
+        }
+      });
+
+      util.itMakesCorrectRequestResponse({
+        title: 'allows verification with autoPush as a function',
+        setup: {
+          status: 'mfa-challenge-sms',
+          request: {
+            uri: '/api/v1/authn/factors/smsigwDlH85L9FyQK0g3/verify?autoPush=true',
+            data: {
+              stateToken: '00rt1IY9c6Q3RVc4a2jJPbS2uAtFNWJz_d8A26KTdF',
+              passCode: '123456'
+            }
+          },
+          response: 'success'
+        },
+        execute: function (test) {
+          return test.trans.verify({
+            passCode: '123456',
+            autoPush: function () {
+              return true;
```

Comment:
What should happen if `autoPush()` doesn't return `true` or `false` (let's say they accidentally return a Promise)?

Reply:
Yea, that was something i was thinking about today. It would probably be better to force whatever the function returns to a boolean.

Reply:
Yeah, let's force a boolean